tsp/src/crypto/nonconfidential.rs

```diff
@@ -37,7 +37,14 @@ pub fn sign(
 pub fn verify<'a>(
     sender: &dyn VerifiedVid,
     tsp_message: &'a mut [u8],
-) -> Result<&'a [u8], CryptoError> {
+) -> Result<
+    (
+        &'a [u8],
+        crate::cesr::CryptoType,
+        crate::cesr::SignatureType,
```

Comment:
- Is it possible to use `MessageType` here? (Or is there a reason why that is not preferable?)

- Is the cryptotype for `verify` not always "plaintext"?

Reply:
I agree, using a `MessageType` would be nicer here.

The crypto type should indeed always be "plaintext" here, as otherwise the `ciphertext` would not be `None` and it would return a crypto error. We could hard code the crypto type to make this more explicit, but I prefer using the values from the packet's envelope for future maintainability.